bitnami/mariadb/templates/primary/svc.yaml

```
@@ -19,24 +19,32 @@ metadata:
     {{- include "common.tplvalues.render" ( dict "value" .Values.metrics.annotations "context" $ ) | nindent 4 }}
     {{- end }}
 spec:
-  type: {{ .Values.primary.service.type }}
-  {{- if and (eq .Values.primary.service.type "ClusterIP") .Values.primary.service.clusterIP }}
+  {{- if and .Values.primary.service.clusterIP (eq .Values.primary.service.type "ClusterIP") }}
   clusterIP: {{ .Values.primary.service.clusterIP }}
   {{- end }}
-  {{- if and .Values.primary.service.loadBalancerIP (eq .Values.primary.service.type "LoadBalancer") }}
+  {{- if (or (eq .Values.primary.service.type "LoadBalancer") (eq .Values.primary.service.type "NodePort")) }}
+  externalTrafficPolicy: {{ .Values.primary.service.externalTrafficPolicy | quote }}
+  {{- end }}
+  {{ if eq .Values.primary.service.type "LoadBalancer" }}
+  loadBalancerSourceRanges: {{ .Values.primary.service.loadBalancerSourceRanges }}
+  {{ end }}
```

This change is not correct. What we had previously was more accurate:
```suggestion
  {{- if and (eq .Values.primary.service.type "LoadBalancer") .Values.primary.service.loadBalancerSourceRanges }}
  loadBalancerSourceRanges: {{ .Values.primary.service.loadBalancerSourceRanges }}
  {{ end }}
```

---

I'm sorry, but I copied this exactly as in the Template chart:
https://github.com/bitnami/charts/blob/e34ee5c5aeafabe15cbf0d2b0fecb18483bc6415/template/CHART_NAME/templates/service.yaml#L19-L21

Should I apply the suggested change anyway? We are configuring `loadBalancerSourceRanges` as in this PR in many other charts.

---

Then the template is wrong @migruiz4 😄 :
```console
$ k explain service.spec.loadBalancerSourceRanges
...
If specified and supported by the platform, this will restrict traffic
through the cloud-provider load-balancer will be restricted to the
specified client IPs
```
It doesn't make sense to indicate this setting UNLESS the service type is actually "LoadBalancer"